Source/WebKit/Platform/IPC/Connection.cpp

```diff
@@ -1356,7 +1353,18 @@ CompletionHandler<void(Decoder*)> takeAsyncReplyHandler(Connection& connection,
 
 void Connection::wakeUpRunLoop()
 {
-    RunLoop::main().wakeUp();
+    if (!isValid())
+        return;
+    m_syncState->runLoop().wakeUp();
+}
+
+template<typename F>
+void Connection::dispatchClient(F&& clientRunLoopTask)
+{
+    Locker lock { m_incomingMessagesLock };
+    if (!m_syncState)
+        return;
+    m_syncState->runLoop().dispatch(WTFMove(clientRunLoopTask));
```

Comment:
Confusing to be getting a runLoop from m_syncState when this code is not about sync IPC